tools/coverage/jacoco.gradle

```diff
@@ -0,0 +1,69 @@
+apply from: "$rootDir/tools/coverage/jacoco-common.gradle"
+
+afterEvaluate {
+
+    if (!project.hasProperty('android')) return
+
+    def jacocoTasks = []
+    def jacocoVerificationTasks = []
+
+    android.productFlavors.all {
+        flavor ->
+            jacocoTasks.add("jacocoTestReport-${flavor.name}")
+            jacocoVerificationTasks.add("jacocoTestCoverageVerification-${flavor.name}")
+    }
+
+    if (android.productFlavors.isEmpty()) {
+        jacocoTasks.add("jacocoTestReport")
+        jacocoVerificationTasks.add("jacocoTestCoverageVerification")
+    }
+
+    jacocoTasks.each { jacocoTask ->
+        task(jacocoTask, type: JacocoReport, group: "coverage-report") {
+            def flavorName = jacocoTask.split("-").last()
+            flavorName = flavorName == jacocoTask ? "" : flavorName
+
+            // Run Tests
+            if (project.android.buildTypes.debug.testCoverageEnabled)
+                dependsOn "create${flavorName.capitalize()}DebugCoverageReport"
+            dependsOn "test${flavorName.capitalize()}DebugUnitTest"
+
+            reports {
+                xml.enabled = true
+                html.enabled = true
+            }
+
+            sourceDirectories = files(findSourceDirectories(project, flavorName))
+            classDirectories = files(findClassDirectories(project, flavorName))
+            executionData = fileTree(dir: "$buildDir", includes: findCoverageFiles(flavorName))
+
+            doFirst {
+                println files(findClassDirectories(project, flavorName)).files
+            }
+
+            doLast {
+                println "Coverage report path -> file://${reports.html.destination}/index.html\n"
+            }
+        }
+    }
+
+    jacocoVerificationTasks.each { jacocoVerificationTask ->
+        task(jacocoVerificationTask, type: JacocoCoverageVerification, group: "coverage-report") {
+            def flavorName = jacocoVerificationTask.split("-").last()
+            flavorName = flavorName == jacocoVerificationTask ? "" : flavorName
+
+            violationRules {

```

Comment:
This are the same rules from `jacoco-unified.gradle`? Do you think that worth the effort to extract in a separated method? 
For a future version, it would be nice if the user can define the value of `failOnViolation`.

Reply:
The task is not attached on any other task, so the idea was run this task only when you need to see if your code coverage is ok. But we can think about something in the future, like fail the task only when we are building the release variant and attach the verification with the coverage task.

We can even seek some way to put some system environments to build some custom rule on Continuous Integration or add some other violationRules

Reply:
But for now, I think it's ok. =P